pandas/tests/indexes/test_base.py

```diff
@@ -678,61 +678,53 @@ def test_intersection(self):
         inter = first.intersection(first)
         assert inter is first
 
+    @pytest.mark.parametrize("idx2,keeps_nm", [
+        (Index([3, 4, 5, 6, 7], name="idx"), True),  # preserve same name
+        (Index([3, 4, 5, 6, 7], name="other"), False),  # drop diff names
+        (Index([3, 4, 5, 6, 7]), False)])
+    def test_intersection_name_preservation(self, idx2, keeps_nm):
         idx1 = Index([1, 2, 3, 4, 5], name='idx')
-        # if target has the same name, it is preserved
-        idx2 = Index([3, 4, 5, 6, 7], name='idx')
-        expected2 = Index([3, 4, 5], name='idx')
-        result2 = idx1.intersection(idx2)
-        tm.assert_index_equal(result2, expected2)
-        assert result2.name == expected2.name
-
-        # if target name is different, it will be reset
-        idx3 = Index([3, 4, 5, 6, 7], name='other')
-        expected3 = Index([3, 4, 5], name=None)
-        result3 = idx1.intersection(idx3)
-        tm.assert_index_equal(result3, expected3)
-        assert result3.name == expected3.name
-
-        # non monotonic
-        idx1 = Index([5, 3, 2, 4, 1], name='idx')
-        idx2 = Index([4, 7, 6, 5, 3], name='idx')
-        expected = Index([5, 3, 4], name='idx')
-        result = idx1.intersection(idx2)
-        tm.assert_index_equal(result, expected)
-
-        idx2 = Index([4, 7, 6, 5, 3], name='other')
-        expected = Index([5, 3, 4], name=None)
-        result = idx1.intersection(idx2)
-        tm.assert_index_equal(result, expected)
+        exp = Index([3, 4, 5])
+        res = idx1.intersection(idx2)
 
-        # non-monotonic non-unique
-        idx1 = Index(['A', 'B', 'A', 'C'])
-        idx2 = Index(['B', 'D'])
-        expected = Index(['B'], dtype='object')
-        result = idx1.intersection(idx2)
-        tm.assert_index_equal(result, expected)
+        if keeps_nm:
+            exp.name = 'idx'
 
-        idx2 = Index(['B', 'D', 'A'])
-        expected = Index(['A', 'B', 'A'], dtype='object')
-        result = idx1.intersection(idx2)
-        tm.assert_index_equal(result, expected)
+        assert res.name == exp.name
+        tm.assert_index_equal(res, exp)
 
-        # preserve names
+    @pytest.mark.parametrize("first_nm,sec_nm,exp_nm", [
+        ('A', 'A', 'A'), ('A', 'B', None), (None, 'B', None)])
+    def test_intersection_name_preservation2(self, first_nm, sec_nm, exp_nm):
         first = self.strIndex[5:20]
         second = self.strIndex[:10]
-        first.name = 'A'
-        second.name = 'A'
+        first.name = first_nm
+        second.name = sec_nm
         intersect = first.intersection(second)
-        assert intersect.name == 'A'
+        assert intersect.name == exp_nm
 
-        second.name = 'B'
-        intersect = first.intersection(second)
-        assert intersect.name is None
+    @pytest.mark.parametrize("idx2,keeps_nm", [
```

Comment:
so ``keeps_nm`` -> ``keeps_name`` and normalize the input params, e.g. ``index`` (or index2 or other) as appropriate